modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -315,8 +329,14 @@ func (p *Processor) spanFailed(span *v1_trace.Span) bool {
 	return span.GetStatus().GetCode() == v1_trace.Status_STATUS_CODE_ERROR
 }
 
+func unixNanosDiffSec(unixNanoStart uint64, unixNanoEnd uint64) float64 {
+	// handling potential underflow of unit64s substraction
+	diff := int64(unixNanoEnd) - int64(unixNanoStart)
```

Comment:
instead of casting we could check to see if diff > end for overflow. is that better?

Reply:
also, we recently made a change to protect span metrics from negative latencies:

https://github.com/grafana/tempo/pull/3412/files

@mdisibio should we make sure the same protections are here?

Reply:
> instead of casting we could check to see if diff > end for overflow. is that better?

I agree we can make it cleaner,
Really liked your idea, very clever 🙃 My only concern about it is how clear it will be. After making the check if diff > end, if underflow did happen, meaning the subtraction yields a negative number - don't we want to return it?

If not, maybe we could make this function a bit more specific and move the negative values "protection" into it.. and then.. return 0?
It could also be that simple:
```
func unixNanosDiffSec(unixNanoStart uint64, unixNanoEnd uint64) float64 {
	if unixNanoStart > unixNanoEnd {
		// To prevent underflow, return 0.
		return 0
	}
	// Safe subtraction.
	return float64(unixNanoEnd-unixNanoStart) / 1e9
}
```

Reply:
let's follow suit with the linked PR for consistency and return 0.

i think your suggested `unixNanosDiffSec` in this PR is perfect.

Reply:
so this one is resolved? 🤔